### PR TITLE
fix(files_sharing): fix mtime update on sharing views

### LINKED ISSUE #44026: [Bug]: Share views show everything as modified a few seconds ago
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Bildschirmfoto vom 2024-03-06 12-09-49

### Steps to reproduce

1. Open `/apps/files/sharingout`

### Expected behavior

Real modified dates

### Installation method

None

### Nextcloud Server version

28

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix https://github.com/nextcloud/server/issues/44026

### CODE REVIEW

File: apps/files_sharing/src/services/SharingService.ts [hunk 1]
```diff
@@ -178,7 +178,16 @@ export const getContents = async (sharedWithYou = true, sharedWithOthers = true,
...
-		node.attributes['share-types'] = nodes.map(node => node.attributes['share-types'])
+		const mtime = node.mtime
+		const shareTypes = nodes.map(node => node.attributes['share-types'])
+
+		// Updating attributes will reset the mtime
+		node.attributes['share-types'] = shareTypes
+
+		// Restore mtime to the most recent share
+		// @ts-expect-error _data is private, mtime is readonly
+		node._data.mtime = mtime
```
Comment (artonge) [inline apps/files_sharing/src/services/SharingService.ts:189]: Should we have an API for that in the future?

**Review by artonge — APPROVED** [review]

File: apps/files_sharing/src/services/SharingService.ts (hunk 1 above)
Comment (skjnldsv) [reply apps/files_sharing/src/services/SharingService.ts:189]: Not really, mtime aren't really supposed to manually be updated.
They are automatically set to the current time on specific events, but that's it.

here it's more like a logic issue, and I am avoiding to much of a code change for backport reasons.
The solution would be to create the `Node` after grouping, but I wanna be simpler here

**Review by susnux — APPROVED** [review]
Maybe we should not always update the mtime on the node when adjusting attributes?

**Comment by skjnldsv** [comment]
Closing in favour of https://github.com/nextcloud-libraries/nextcloud-files/pull/974

Then we can release and just update up to stable28 :+1:

**Pull request closed without merging**